graphql/netflix_dgs/src/main/java/com/example/netflixdgs/AppConfig.java

```diff
@@ -1,21 +1,51 @@
 package com.example.netflixdgs;
 
-import java.net.InetSocketAddress;
 
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.http.client.SimpleClientHttpRequestFactory;
 import org.springframework.web.client.RestTemplate;
-import java.net.Proxy;
-import java.net.Proxy.Type;
+
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+
+import java.util.concurrent.TimeUnit;
+
+import org.apache.hc.client5.http.ConnectionKeepAliveStrategy;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager;
+import org.apache.hc.client5.http.impl.routing.DefaultProxyRoutePlanner;
+import org.apache.hc.core5.http.HttpHost;
+import org.apache.hc.core5.http.HttpResponse;
+import org.apache.hc.core5.http.protocol.HttpContext;
+import org.apache.hc.core5.util.TimeValue;
 
 @Configuration
 public class AppConfig {
+    
     @Bean
     public RestTemplate restTemplate() {
-			  SimpleClientHttpRequestFactory factory = new SimpleClientHttpRequestFactory();
-				Proxy proxy = new Proxy(Type.HTTP, new InetSocketAddress("localhost", 3000));
-        factory.setProxy(proxy);
+        final PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+        // the maximum number of connections. See https://github.com/tailcallhq/graphql-benchmarks/blob/main/wrk/bench.sh
+        connectionManager.setMaxTotal(100); 
+        connectionManager.setDefaultMaxPerRoute(20);
+        
+        final ConnectionKeepAliveStrategy keepAliveStrategy = new ConnectionKeepAliveStrategy() {
+            public TimeValue getKeepAliveDuration(HttpResponse response, HttpContext context) {
+            	// Keep alive for 5 seconds
+                return TimeValue.of(5, TimeUnit.SECONDS);
+            }
```

Comment:
Try different settings for KeepAlive and ConnectionManager to see if the performance improves.